server/service/src/document/document_service.rs

```diff
@@ -30,6 +31,45 @@ impl From<RepositoryError> for DocumentHistoryError {
     }
 }
 
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct DocumentDelete {
+    pub id: String,
+    pub patient_id: String,
+    pub parent: String,
+    pub comment: Option<String>,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum DocumentDeleteError {
+    DatabaseError(RepositoryError),
+    InternalError(String),
+}
+
+impl From<RepositoryError> for DocumentDeleteError {
+    fn from(err: RepositoryError) -> Self {
+        DocumentDeleteError::DatabaseError(err)
+    }
+}
+
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct DocumentUndelete {
+    pub id: String,
+    pub parent: String,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum DocumentUndeleteError {
+    CannotUndeleteDocument,
```

Comment:
Could pass in a String saying document hasn't been deleted? 🤔

Reply:
Are there any more errors you want to include for DocumentDelete or DocumentUndelete?

Reply:
Maybe DocumentDoesNotExists if the specified id is invalid